packages/cli/src/commands/governance/vote.ts

```diff
@@ -22,21 +23,25 @@ export default class Vote extends BaseCommand {
   ]
 
   async run() {
-    const kit = await this.getKit()
     const res = await this.parse(Vote)
     const signer = res.flags.from
     const id = res.flags.proposalID
-    const voteValue = res.flags.value as keyof typeof VoteValue
-
-    kit.defaultAccount = signer
-    const governance = await kit.contracts.getGovernance()
+    const voteValue = res.flags.value
 
     await newCheckBuilder(this, signer)
       .isVoteSignerOrAccount()
       .proposalExists(id)
       .proposalInStage(id, 'Referendum')
       .runChecks()
 
-    await displaySendTx('voteTx', await governance.vote(id, voteValue), {}, 'ProposalVoted')
+    const walletClient = await this.getWalletClient()
+    const publicClient = await this.getPublicClient()
+
+    // do not wait this. it will be awiated in the displayViemTx
+    const pendingVote = vote(walletClient, id, voteValue)
```

Comment:
We can make displaySendTx accept both promises and txhashes if this is awkward to use.

Reply:
no i love it how it is.